File: services/core-api/app/schemas/support.py
"""Schemas for support requests."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class SupportContext(BaseModel):
    """Context captured with support request."""

    page_url: str
    timestamp: datetime
    user_agent: str
    legacy_id: str | None = None
    session_duration_seconds: int | None = None
    recent_errors: list[str] = Field(default_factory=list)


class SupportRequestCreate(BaseModel):
    """Request to create a support ticket."""

    category: str = Field(
        ...,
        pattern="^(general_question|bug_report|feature_request|account_issue|other)$",
        description="Support request category",
    )
    subject: str = Field(..., min_length=1, max_length=100)
    message: str = Field(..., min_length=1, max_length=2000)
    context: SupportContext


class SupportRequestResponse(BaseModel):
    """Response after creating support request."""

    id: UUID
    category: str
    subject: str
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: apps/web/src/lib/api/support.ts
/**
 * Support API client for help requests.
 */

import { apiPost } from './client';

// Types
export interface SupportContext {
  page_url: string;
  timestamp: string;
  user_agent: string;
  legacy_id: string | null;
  session_duration_seconds: number | null;
  recent_errors: string[];
}

export interface SupportRequestCreate {
  category:
    | 'general_question'
    | 'bug_report'
    | 'feature_request'
    | 'account_issue'
    | 'other';
  subject: string;
  message: string;
  context: SupportContext;
}

export interface SupportRequestResponse {
  id: string;
  category: string;
  subject: string;
  status: string;
  created_at: string;
}

// API Functions
export async function createSupportRequest(
  data: SupportRequestCreate
): Promise<SupportRequestResponse> {
  return apiPost<SupportRequestResponse>('/api/support/request', data);
}
